services/frontend/adapters/inbound/k8s_sidecar_notifications_receiver.adapter.go

```diff
@@ -0,0 +1,131 @@
+package inboundAdapters
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/berops/claudie/services/frontend/domain/usecases"
+)
+
+// Receives notifications from the k8s-sidecar container running in the same pod
+// The notifications are regarding new/updated claudie manifests
+type K8sSidecarNotificationsReceiver struct {
+	usecases *usecases.Usecases
+
+	// manifestDir represents the path to the directory containing claudie manifest files which will be
+	// watched by the k8s-sidecar container.
+	manifestDir string
+
+	// server is the underlying HTTP server which receives notifications
+	// from the k8s sidecar container when the manifest files are created/updated/deleted.
+	server *http.Server
+
+	// waitGroup is used to handle a graceful shutdown of the HTTP server.
+	// It will wait for all spawned go-routines to finish their work.
+	waitGroup sync.WaitGroup
+}
+
+// NewK8sSidecarNotificationsReceiver creates an instance of the K8sSidecarNotificationsReceiver struct.
+// It registers the notification handling route to the underlying HTTP server.
+// Then performs an initial healthcheck of K8sSidecarNotificationsReceiver.
+func NewK8sSidecarNotificationsReceiver(usecases *usecases.Usecases) (*K8sSidecarNotificationsReceiver, error) {
+	manifestDir, isEnvFound := os.LookupEnv("MANIFEST_DIR")
+	if !isEnvFound {
+		return nil, fmt.Errorf("env MANIFES_DIR not found")
+	}
```

Comment:
If we're passing the `usecases` here shouldn't we take the `manifest` dir from it instead of calling `os.LookupEnv` ?

move the 

```go
manifestDir, isEnvFound := os.LookupEnv("MANIFEST_DIR")
	if !isEnvFound {
		return nil, fmt.Errorf("env MANIFES_DIR not found")
	}
```

into the `run` function in the `main.go` function

Reply:
**usecases** contains only the business logic. It will be better if we `don't couple the manifestDir with the business logic layer`.